src/basic/strv.c

```diff
@@ -994,6 +994,42 @@ char **strv_parse_password(const char *s, size_t l) {
         return v;
 }
 
+/* The equivalent of strv_make_nulstr, but for sensitive data. */
+int strv_to_password(char **l, char **p, size_t *q) {
```

Comment:
Same comment here: let's make strv_make_nulstr() optionally support this behaviour.